nrf_modem/doc/pdn_extension.rst

```diff
@@ -157,21 +157,43 @@ Waiting for an IPv6 connection
 
    at_fd = nrf_socket(NRF_AF_LTE, 0, NRF_NPROTO_AT);
    nrf_send(at_fd, "AT+CGEREP=1", strlen("AT+CGEREP=1"));
-   # -> AT reponse "OK"
+   nrf_recv(at_fd, buf, sizeof(buf), 0);
+
+   if (strcmp("OK", buf) == 0) {
+        /* If the command did not return OK we failed to subscribe on the
+         * packet domain notifications.
+         */
+        nrf_close(at_fd);
+        return -1;
+   }
 
    # Set up PDN socket using the APN (PDN initialize and connect)
 
    nrf_getsockopt(pdn_fd, NRF_SOL_PDN, NRF_SO_PDN_CONTEXT_ID, &context_id, sizeof(uint8_t));
-   while not (context_id in ":CGEV IPV6 <cid>");
+
+   char pdn_notification[20];
+   snprintf(pdn_notification, 20, "+CGEV IPV6 %d", context_id);
+
+   int max_bytes;
+   do {
+        int recv_data = nrf_recv(at_fd, buf, sizeof(buf), 0);
+        if(recv_data < 0) {
+                /* nrf_recv failed for reason, this should be handled */
+                return -1;
+        }
+        max_bytes = MIN(recv_data, sizeof(pdn_notification));
+   } while(strncmp(buf, max_bytes, pdn_notification) != 0)
+
    nrf_close(at_fd);
 
-When requesting an IPv6 connection (which requires PDN type IPV6 or IPV4V6), the IPv6 address might not be correct even if the request is successful.
+When requesting an IPv6 connection (which requires PDN type IPV6 or IPV4V6), the IPv6 address might not be global even if the request is successful.
 The reason for this is that many networks use SLAAC, where the prefix of the IPv6 address is set after the PDN connection is created.
-Therefore, you might need to wait with using an IPv6 socket until the prefix is properly set.
-If the address cannot be obtained, a ``+CGEV: IPV6 FAIL <cid>`` error is returned.
+Therefore, you might need to wait before using an IPv6 socket until the prefix is properly set.
+If the address is obtained, a ``+CGEV: IPV6 <cid>`` notification is received from the AT socket, if not a ``+CGEV: IPV6 FAIL <cid>`` notification is received.
```

Comment:
```suggestion
If the address can be obtained, the AT socket sends a ``+CGEV: IPV6 <cid>`` notification.
Otherwise, it sends a ``+CGEV: IPV6 FAIL <cid>`` notification.
```